src/qibo/backends/tensorflow.py

```diff
@@ -290,11 +290,18 @@ def density_matrix_half_matrix_call(self, gate, state):
                             2 * gate.nqubits, *gate.target_qubits,
                             self.get_threads())
 
+    def _result_tensor(self, result):
+        n = len(result)
+        result = sum(2 ** (n - i - 1) * r for i, r in enumerate(result))
+        return self.cast(result, dtype="DTYPEINT")
```

Comment:
Does this operation is strictly necessary?

Reply:
I think yes, because the custom operators for collapse expect the result to be integer (in decimal form) while the other backends use the array form which represents the same number in binary. Note that this is a new addition here, we already do exactly the same conversion inside `core.gates.M` (see [here](https://github.com/qiboteam/qibo/blob/b706ed3c014f139ff43794bef03db9e464f28774/src/qibo/core/gates.py#L250) in master). I just moved the conversion at the backend level in this PR because it is slightly different for qibojit and qibotf so it is easier to implement this way.

There may be a way to avoid this conversion by using decimal as the default representation for `result` and convert to binary when using the numpy/tensorflow backends.